transcoder/src/metadata.go

```diff
@@ -64,13 +119,31 @@ func NewMetadataService() (*MetadataService, error) {
 	}
 	m.Up()
 
-	return &MetadataService{
-		database:     db,
-		lock:         NewRunLock[string, *MediaInfo](),
-		thumbLock:    NewRunLock[string, interface{}](),
-		extractLock:  NewRunLock[string, interface{}](),
-		keyframeLock: NewRunLock[KeyframeKey, *Keyframe](),
-	}, nil
+	return db, nil
+}
+
+func (s *MetadataService) setupStorage(ctx context.Context) (storage.StorageBackend, error) {
+	s3BucketName := os.Getenv("S3_BUCKET_NAME")
```

Comment:
Any opinion on whether or not I should add an explicit `BACKEND_TYPE` env var here? In the off chance that another backend is added at some point (such as some form of document storage), this could make the change + migrating significantly easier.

Reply:
both are fines to me

Reply:
I'll leave it as is for now, so that it remains consistent with the other services